modules/apps/depot/depot-test/src/testIntegration/java/com/liferay/depot/service/test/DepotEntryDLAppServiceWhenCopyingWithDocumentTypesTest.java

```diff
@@ -0,0 +1,406 @@
+
+import java.util.HashMap;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Alicia García
+ */
+@RunWith(Arquillian.class)
+public class DepotEntryDLAppServiceWhenCopyingWithDocumentTypesTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_depotEntry = _depotEntryLocalService.addDepotEntry(
+			HashMapBuilder.put(
+				LocaleUtil.getDefault(), "name"
+			).build(),
+			HashMapBuilder.put(
+				LocaleUtil.getDefault(), "description"
+			).build(),
+			ServiceContextTestUtil.getServiceContext());
+
+		_depotGroup = _groupLocalService.getGroup(_depotEntry.getGroupId());
+
+		DDMStructure depotDDMStructure = _ddmStructureLocalService.addStructure(
+			_depotGroup.getCreatorUserId(), _depotGroup.getGroupId(),
+			DDMStructureConstants.DEFAULT_PARENT_STRUCTURE_ID,
+			PortalUtil.getClassNameId(DLFileEntryMetadata.class),
+			StringPool.BLANK,
+			HashMapBuilder.put(
+				LocaleUtil.getDefault(),
+				DLFileEntryMetadata.class.getSimpleName()
+			).build(),
+			new HashMap<>(), StringPool.BLANK, StorageType.DEFAULT.toString(),
+			ServiceContextTestUtil.getServiceContext(_depotGroup.getGroupId()));
+
+		_depotDLFileEntryType = _dlFileEntryTypeLocalService.addFileEntryType(
+			_depotGroup.getCreatorUserId(), _depotGroup.getGroupId(),
+			depotDDMStructure.getStructureId(), null,
+			HashMapBuilder.put(
+				LocaleUtil.getDefault(), RandomTestUtil.randomString()
+			).build(),
+			new HashMap<>(),
+			DLFileEntryTypeConstants.FILE_ENTRY_TYPE_SCOPE_DEFAULT,
+			ServiceContextTestUtil.getServiceContext(_depotGroup.getGroupId()));
+
+		_group = GroupTestUtil.addGroup();
+
+		DDMStructure ddmStructure = _ddmStructureLocalService.addStructure(
+			_group.getCreatorUserId(), _group.getGroupId(),
+			DDMStructureConstants.DEFAULT_PARENT_STRUCTURE_ID,
+			PortalUtil.getClassNameId(DLFileEntryMetadata.class),
+			StringPool.BLANK,
+			HashMapBuilder.put(
+				LocaleUtil.getDefault(),
+				DLFileEntryMetadata.class.getSimpleName()
+			).build(),
+			new HashMap<>(), StringPool.BLANK, StorageType.DEFAULT.toString(),
+			ServiceContextTestUtil.getServiceContext(_group.getGroupId()));
+
+		_dlFileEntryType = _dlFileEntryTypeLocalService.addFileEntryType(
+			_group.getCreatorUserId(), _group.getGroupId(),
+			ddmStructure.getStructureId(), null,
+			HashMapBuilder.put(
+				LocaleUtil.getDefault(), RandomTestUtil.randomString()
+			).build(),
+			new HashMap<>(),
+			DLFileEntryTypeConstants.FILE_ENTRY_TYPE_SCOPE_DEFAULT,
+			ServiceContextTestUtil.getServiceContext(_group.getGroupId()));
+	}
+
+	@Test(expected = InvalidFileEntryTypeException.class)
+	public void testCopyFileEntryFailsUnlessDLFileEntryTypeAvailable()
```

Comment:
I used the wrong word, sorry. "Ambiguity" is not probably what I meant. What I want to say is: it is clearer, at least to me, to read a sentence that uses standard words "When" and "Not" instead of "unless". Therefore, this is an optional change.